californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/ObserveClientSideTest.java

```diff
@@ -433,6 +435,99 @@ public void testBlockwiseObserveAndNotificationWithoutBlockwise() throws Excepti
 		response = notificationListener.waitForResponse(1000);
 		assertResponseContainsExpectedPayload(response, notifyPayload);
 	}
+	
+	
+	/**
+	 * Verifies behavior of observing a resource which start a blockwise
+	 * transfer and receiving an older none-blockwise notification before previous
+	 * blockwise transfer finished.
+	 * 
+	 * <pre>
+	 *    (actual used MIDs and Tokens my vary!)
+	 *    ####### establish observe an start blockwise tranfer #############
+	 *    CON [MID=37434, T=cbeb45abdb65aba2], GET, /test, observe(2)    ----->
+	 *    <-----   ACK [MID=37434, T=cbeb45abdb65aba2], 2.05, 2:0/1/16, observe(2)
+	 *    CON [MID=37435, T=d2b1c9e97df24976], GET, /test, 2:1/0/16    ----->
+	 *    ####### receive "older" none blockwise notification #############
+	 *    <-----   CON [MID=222, T=cbeb45abdb65aba2], 2.05, observe(0)
+	 *    ACK [MID=222]   ----->
+	 *    ####### ensure previous blockwise tranfer continue #############
+	 *    <-----   ACK [MID=37435, T=d2b1c9e97df24976], 2.05, 2:1/1/16
+	 *    ... response received
+	 *    ####### ensure next none-blockwise notification is received #############
+	 *    <-----   CON [MID=223, T=cbeb45abdb65aba2], 2.05, observe(3)
+	 *    ACK [MID=223]   ----->
+	 *    ####### ensure next blockwise notification is received #############
+	 *    <-----   CON [MID=224, T=cbeb45abdb65aba2], 2.05, 2:0/1/16, observe(4)
+	 *    ACK [MID=224]   ----->
+	 *    CON [MID=21291, T=61c34585da9003e7], GET, /test, 2:1/0/16    ----->
+	 *    <-----   ACK [MID=21291, T=61c34585da9003e7], 2.05, 2:1/0/16
+	 * </pre>
+	 * 
+	 * @throws Exception
+	 */
+	@Test
+	public void testBlockwiseObserveNotInterruptedByOdlerNotificationWithoutBlockwise() throws Exception {
+		System.out.println("Blockwise Observe:");
+		// observer request response will be sent using blockwise
+		respPayload = generateRandomPayload(2 * 16);
+		// notification payload sended without blockwise
+		String notifyPayload = generateRandomPayload(8);
+		String path = "test";
+
+		Request request = createRequest(GET, path, server);
+		request.setObserve();
+		SynchronousNotificationListener notificationListener = new SynchronousNotificationListener(request);
+		client.addNotificationListener(notificationListener);
+
+		// Send observe request
+		client.sendRequest(request);
+		// Expect observe request
+		server.expectRequest(CON, GET, path).storeMID("OBS_REQ").storeToken("OBS_TOK").go();
```

Comment:
Just to say: you can use `storeBoth("ID")` and access the parts with `loadToken("ID")` or `loadMID("ID")`.